pandas/core/groupby/groupby.py

```diff
@@ -726,7 +726,9 @@ def _selected_obj(self):
 
         if self._selection is None or isinstance(self.obj, Series):
             if self._group_selection is not None:
-                return self.obj[self._group_selection]
+                return self.obj._take(
```

Comment:
I think this is the same as _obj_with_exclusions, which should already be cached, so we could avoid making a copy

Reply:
This is great - not only that, but we can also avoid all the code that determines `_group_selection`. I've turned `_group_selection` into a Boolean flag.

Reply:
should we expect this change to affect the timings in the OP?

Reply:
ASVs updated; essentially the same results.